grizzly/common/adapter.py

```diff
@@ -25,21 +43,17 @@ class Adapter(metaclass=ABCMeta):
 
     __slots__ = ("_harness", "fuzz", "monitor", "remaining")
 
-    #############################
-    # Built-ins do NOT overload!
-    #############################
-
     def __init__(self):
         if not isinstance(self.NAME, str):
             raise AdapterError("%s.NAME must be a string" % (type(self).__name__,))
         self._harness = None
         self.fuzz = dict()
         self.monitor = None
-        # remaining can be used to indicate the number of TestCases remaining to process
         self.remaining = None
 
     def cleanup(self):
-        """Automatically called once at shutdown.
+        """Automatically called once at shutdown. Used internally by Grizzly.
+        *** DO NOT OVERLOAD! ***
```

Comment:
So warnings like this feel like a footgun. They're not likely to be noticed, and do I use `cleanup()` or `shutdown()`? Why not use `__cleanup()` to ensure the user can't mess it up? (same for other "private" methods and attributes).

Reply:
Well I guess I was thinking they aren't technically private because they are called by Grizzly.